src/main/java/com/google/sps/servlets/utils/Statistics.java

```diff
@@ -60,6 +71,41 @@ public Statistics(List<UserComment> userCommentList, int topNComments) {
     sentimentBucketList = categorizeToBucketList(userCommentList, topNComments);
     averageScore = getAverageValue(userCommentList, "score");
     averageMagnitude = getAverageValue(userCommentList, "Magnitude");
+    wordFrequencyMap = countWordFrequencyMap(userCommentList);
+  }
+
+  /**
+   * Convert given userCommentList into a word map: {word: frequency}
+   *
+   * @param userCommentList a list of userComment with all fields updated
+   * @return wordFrequencyMap to represent each word appearance time
+   */
+  private Map<String, Integer> countWordFrequencyMap(List<UserComment> userCommentList) {
+    List<String> wordsToIgnore = CommonWordsRetriever.getCommonWords();
+    // Flatten all user comment message into a list of words
+    Stream<String> allWordStream =
+        // Text extractor removes all HTML tags and returns only the text
+        userCommentList.stream()
+            .map(
+                comment ->
+                    new Source(comment.getCommentMsg())
+                        .getTextExtractor()
+                        .toString()
+                        .replaceAll("[^a-zA-Z0-9\\s]", "")
```

Comment:
Do you want to remove numbers as well as it is a word frequency map?

Reply:
Spoke with Xin offline, we want to include numbers since it could be related to the video somehow